internal/assertions/stderr_assertion.go

```diff
@@ -0,0 +1,47 @@
+package assertions
+
+import (
+	"fmt"
+
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+type StderrAssertion struct {
+	ExpectedLines []string
+}
+
+func NewStderrAssertion(expectedLines []string) StderrAssertion {
+	return StderrAssertion{ExpectedLines: expectedLines}
+}
+
+func (a StderrAssertion) Run(stderr []string, logger *logger.Logger) error {
+	var output []string
+
+	for i, expectedLine := range a.ExpectedLines {
+		if i >= len(stderr) {
+			logAllSuccessLogs(output, logger)
+			logger.Errorf("? %s", expectedLine)
+			return fmt.Errorf("Expected line #%d on stderr to be %q, but didn't find line", i+1, expectedLine)
```

Comment:
This is going to be confusing since we're filtering out stderr lines before comparison. Ideas on how to make this more user-friendly?

Example: 

If I omit the trailing `]` and use `[line 0 Error ...`, here it'll say "Expected line #1 to be ..., but didn't find line"

Reply:
Also, the reason I mentioned that we should split out stderr & stdout is that we can handle cases like this well - if both used exactly the same logic, the DRY principle applies and they could've just been one

Reply:
Updated, stderr errors, to make it clearer that we skipped N number of lines that didn't start with `[line N]`.